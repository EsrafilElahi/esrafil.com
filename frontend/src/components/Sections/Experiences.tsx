import { VerticalTimeline } from 'react-vertical-timeline-component';
import 'react-vertical-timeline-component/style.min.css';
import type { Experience } from '../../../types.ts';
import ExperienceItem from '../ExperienceItem.tsx';

type Props = {
  experiences: Experience[];
};

const Experiences = (props: Props) => {
  const { experiences } = props;

  return (
    <VerticalTimeline>
      {experiences.map((experience) => (
        <ExperienceItem key={experience.id} experience={experience} />
      ))}
    </VerticalTimeline>
  );
};

export default Experiences;
