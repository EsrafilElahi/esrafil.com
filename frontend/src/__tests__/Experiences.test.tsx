import { act, cleanup, render, screen, within } from '@testing-library/react';
import { afterEach, beforeEach, describe, test } from 'vitest';
import Experiences from '../components/Sections/Experiences';
import mockExperinces from '../mocks/experiences.json';

beforeEach(() => {
  render(<Experiences experiences={mockExperinces} />);
});

afterEach(() => {
  cleanup();
});

describe('Experiences Component', async () => {
  test('renders correctly', () => {
    const { container } = render(<Experiences experiences={mockExperinces} />);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Experiences experiences={mockExperinces} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('length of experiences is equal to mockExperinces.length', async () => {
    const exps = await screen.findAllByTestId('exp');
    expect(exps).toHaveLength(4);

    const firstExp = exps[0];
    const secondExp = exps[1];
    const thirdExp = exps[2];
    const fourthExp = exps[3];

    expect(within(firstExp).getByText('frontend developer')).toBeInTheDocument();
    expect(within(secondExp).getByText('frontend reactjs | nextjs | typescript developer')).toBeInTheDocument();
    expect(within(thirdExp).getByText('senior frontend developer')).toBeInTheDocument();
    expect(within(fourthExp).getByText('senior frontend engineer')).toBeInTheDocument();
  });
});
