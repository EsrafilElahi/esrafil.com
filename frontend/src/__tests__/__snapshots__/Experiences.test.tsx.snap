// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Experiences Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="vertical-timeline vertical-timeline--animate vertical-timeline--two-columns"
  >
    <div
      class="vertical-timeline-element--work vertical-timeline-element"
      id=""
    >
      <span
        class="vertical-timeline-element-icon is-hidden"
        style="background: rgb(138, 171, 255); color: rgb(255, 255, 255);"
      >
        <svg
          height="800px"
          id="Layer_1"
          viewBox="0 0 392.513 392.513"
          width="800px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <path
              d="M329.261,256.421H63.37c0,0-39.434,83.329-39.628,83.588c-5.172,7.887,0.84,19.911,11.119,19.782 H357.77c9.438,0.453,16.743-10.925,11.119-19.782C368.695,339.75,329.261,256.421,329.261,256.421z"
              style="fill: #FFFFFF;"
            />
            <path
              d="M325.382,43.669c0-6.012-4.848-10.925-10.925-10.925H78.239c-6.012,0-10.925,4.848-10.925,10.925 v190.966h258.069V43.669z"
              style="fill: #FFFFFF;"
            />
          </g>
          <polygon
            points="48.954,338.005 77.204,278.207 315.426,278.207 343.742,338.005 "
            style="fill: #FFC10D;"
          />
          <path
            d="M89.099,54.595v13.059h39.176c6.012,0,10.925,4.848,10.925,10.925 c0,6.012-4.848,10.925-10.925,10.925H89.099v11.507h16.032c6.012,0,10.925,4.848,10.925,10.925c0,6.012-4.848,10.925-10.925,10.925 H89.164v90.053h214.432V54.595H89.099z"
            style="fill: #56ACE0;"
          />
          <g>
            <path
              d="M387.83,329.277l-40.663-85.98V43.669c0-18.036-14.675-32.711-32.711-32.711H78.239 c-18.036,0-32.711,14.675-32.711,32.711v199.564L4.8,329.213c-14.675,26.828,6.853,52.881,30.125,52.299H357.77 C378.586,382.74,402.893,357.269,387.83,329.277z M78.239,32.809h236.283c6.012,0,10.925,4.848,10.925,10.925V234.7H67.313V43.734 C67.313,37.657,72.162,32.809,78.239,32.809z M357.77,359.791H34.925c-10.279,0.129-16.356-11.895-11.119-19.782 c0.129-0.323,39.564-83.588,39.564-83.588h265.891c0,0,39.434,83.329,39.628,83.588C374.578,348.93,367.208,360.243,357.77,359.791 z"
              style="fill: #194F82;"
            />
            <path
              d="M225.826,310.918h-58.893c-6.012,0-10.925,4.849-10.925,10.925s4.848,10.925,10.925,10.925h58.893 c6.012,0,10.925-4.848,10.925-10.925C236.687,315.766,231.774,310.918,225.826,310.918z"
              style="fill: #194F82;"
            />
            <polygon
              points="124.267,180.009 131.637,158.869 138.683,180.009 149.932,180.009 163.055,141.285  151.871,141.285 144.307,164.687 136.679,141.285 126.271,141.285 118.707,164.687 111.144,141.285 100.024,141.285  113.148,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="189.172,180.009 196.542,158.869 203.653,180.009 214.837,180.009 227.96,141.285  216.776,141.285 209.212,164.687 201.584,141.285 191.176,141.285 183.612,164.687 176.049,141.285 164.929,141.285  178.053,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="254.012,180.009 261.317,158.869 268.428,180.009 279.677,180.009 292.8,141.285  281.616,141.285 273.988,164.687 266.424,141.285 256.016,141.285 248.453,164.687 240.889,141.285 229.77,141.285  242.893,180.009  "
              style="fill: #194F82;"
            />
          </g>
        </svg>
      </span>
      <div
        class="vertical-timeline-element-content is-hidden"
        style="background: rgb(33, 150, 243); color: rgb(255, 255, 255);"
      >
        <div
          class="vertical-timeline-element-content-arrow"
          style="border-right: 7px solid rgb(33, 150, 243);"
        />
        <div
          data-testid="exp"
        >
          <h3
            class="vertical-timeline-element-title text-xl font-bold tracking-wide"
          >
            frontend developer
          </h3>
          <h4
            class="vertical-timeline-element-subtitle"
          >
            Aytronik corp
          </h4>
          <p>
            lorem ipsum dolor sit amet, consectetur adip proident et non proident consequat, sed do eiusmod tempor incididunt ut labore et dolore mag
          </p>
        </div>
        <span
          class="vertical-timeline-element-date"
        >
          July 2019 - Agu 2020
        </span>
      </div>
    </div>
    <div
      class="vertical-timeline-element--work vertical-timeline-element"
      id=""
    >
      <span
        class="vertical-timeline-element-icon is-hidden"
        style="background: rgb(138, 171, 255); color: rgb(255, 255, 255);"
      >
        <svg
          height="800px"
          id="Layer_1"
          viewBox="0 0 392.513 392.513"
          width="800px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <path
              d="M329.261,256.421H63.37c0,0-39.434,83.329-39.628,83.588c-5.172,7.887,0.84,19.911,11.119,19.782 H357.77c9.438,0.453,16.743-10.925,11.119-19.782C368.695,339.75,329.261,256.421,329.261,256.421z"
              style="fill: #FFFFFF;"
            />
            <path
              d="M325.382,43.669c0-6.012-4.848-10.925-10.925-10.925H78.239c-6.012,0-10.925,4.848-10.925,10.925 v190.966h258.069V43.669z"
              style="fill: #FFFFFF;"
            />
          </g>
          <polygon
            points="48.954,338.005 77.204,278.207 315.426,278.207 343.742,338.005 "
            style="fill: #FFC10D;"
          />
          <path
            d="M89.099,54.595v13.059h39.176c6.012,0,10.925,4.848,10.925,10.925 c0,6.012-4.848,10.925-10.925,10.925H89.099v11.507h16.032c6.012,0,10.925,4.848,10.925,10.925c0,6.012-4.848,10.925-10.925,10.925 H89.164v90.053h214.432V54.595H89.099z"
            style="fill: #56ACE0;"
          />
          <g>
            <path
              d="M387.83,329.277l-40.663-85.98V43.669c0-18.036-14.675-32.711-32.711-32.711H78.239 c-18.036,0-32.711,14.675-32.711,32.711v199.564L4.8,329.213c-14.675,26.828,6.853,52.881,30.125,52.299H357.77 C378.586,382.74,402.893,357.269,387.83,329.277z M78.239,32.809h236.283c6.012,0,10.925,4.848,10.925,10.925V234.7H67.313V43.734 C67.313,37.657,72.162,32.809,78.239,32.809z M357.77,359.791H34.925c-10.279,0.129-16.356-11.895-11.119-19.782 c0.129-0.323,39.564-83.588,39.564-83.588h265.891c0,0,39.434,83.329,39.628,83.588C374.578,348.93,367.208,360.243,357.77,359.791 z"
              style="fill: #194F82;"
            />
            <path
              d="M225.826,310.918h-58.893c-6.012,0-10.925,4.849-10.925,10.925s4.848,10.925,10.925,10.925h58.893 c6.012,0,10.925-4.848,10.925-10.925C236.687,315.766,231.774,310.918,225.826,310.918z"
              style="fill: #194F82;"
            />
            <polygon
              points="124.267,180.009 131.637,158.869 138.683,180.009 149.932,180.009 163.055,141.285  151.871,141.285 144.307,164.687 136.679,141.285 126.271,141.285 118.707,164.687 111.144,141.285 100.024,141.285  113.148,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="189.172,180.009 196.542,158.869 203.653,180.009 214.837,180.009 227.96,141.285  216.776,141.285 209.212,164.687 201.584,141.285 191.176,141.285 183.612,164.687 176.049,141.285 164.929,141.285  178.053,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="254.012,180.009 261.317,158.869 268.428,180.009 279.677,180.009 292.8,141.285  281.616,141.285 273.988,164.687 266.424,141.285 256.016,141.285 248.453,164.687 240.889,141.285 229.77,141.285  242.893,180.009  "
              style="fill: #194F82;"
            />
          </g>
        </svg>
      </span>
      <div
        class="vertical-timeline-element-content is-hidden"
        style="background: rgb(33, 150, 243); color: rgb(255, 255, 255);"
      >
        <div
          class="vertical-timeline-element-content-arrow"
          style="border-right: 7px solid rgb(33, 150, 243);"
        />
        <div
          data-testid="exp"
        >
          <h3
            class="vertical-timeline-element-title text-xl font-bold tracking-wide"
          >
            frontend reactjs | nextjs | typescript developer
          </h3>
          <h4
            class="vertical-timeline-element-subtitle"
          >
            Shams uni
          </h4>
          <p>
            lorem ipsum dolor sit amet, consectetur adip proident et non proident consequat, sed do eiusmod tempor incididunt ut labore et dolore mag
          </p>
        </div>
        <span
          class="vertical-timeline-element-date"
        >
          Agu 2020 - Feb 2021
        </span>
      </div>
    </div>
    <div
      class="vertical-timeline-element--work vertical-timeline-element"
      id=""
    >
      <span
        class="vertical-timeline-element-icon is-hidden"
        style="background: rgb(138, 171, 255); color: rgb(255, 255, 255);"
      >
        <svg
          height="800px"
          id="Layer_1"
          viewBox="0 0 392.513 392.513"
          width="800px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <path
              d="M329.261,256.421H63.37c0,0-39.434,83.329-39.628,83.588c-5.172,7.887,0.84,19.911,11.119,19.782 H357.77c9.438,0.453,16.743-10.925,11.119-19.782C368.695,339.75,329.261,256.421,329.261,256.421z"
              style="fill: #FFFFFF;"
            />
            <path
              d="M325.382,43.669c0-6.012-4.848-10.925-10.925-10.925H78.239c-6.012,0-10.925,4.848-10.925,10.925 v190.966h258.069V43.669z"
              style="fill: #FFFFFF;"
            />
          </g>
          <polygon
            points="48.954,338.005 77.204,278.207 315.426,278.207 343.742,338.005 "
            style="fill: #FFC10D;"
          />
          <path
            d="M89.099,54.595v13.059h39.176c6.012,0,10.925,4.848,10.925,10.925 c0,6.012-4.848,10.925-10.925,10.925H89.099v11.507h16.032c6.012,0,10.925,4.848,10.925,10.925c0,6.012-4.848,10.925-10.925,10.925 H89.164v90.053h214.432V54.595H89.099z"
            style="fill: #56ACE0;"
          />
          <g>
            <path
              d="M387.83,329.277l-40.663-85.98V43.669c0-18.036-14.675-32.711-32.711-32.711H78.239 c-18.036,0-32.711,14.675-32.711,32.711v199.564L4.8,329.213c-14.675,26.828,6.853,52.881,30.125,52.299H357.77 C378.586,382.74,402.893,357.269,387.83,329.277z M78.239,32.809h236.283c6.012,0,10.925,4.848,10.925,10.925V234.7H67.313V43.734 C67.313,37.657,72.162,32.809,78.239,32.809z M357.77,359.791H34.925c-10.279,0.129-16.356-11.895-11.119-19.782 c0.129-0.323,39.564-83.588,39.564-83.588h265.891c0,0,39.434,83.329,39.628,83.588C374.578,348.93,367.208,360.243,357.77,359.791 z"
              style="fill: #194F82;"
            />
            <path
              d="M225.826,310.918h-58.893c-6.012,0-10.925,4.849-10.925,10.925s4.848,10.925,10.925,10.925h58.893 c6.012,0,10.925-4.848,10.925-10.925C236.687,315.766,231.774,310.918,225.826,310.918z"
              style="fill: #194F82;"
            />
            <polygon
              points="124.267,180.009 131.637,158.869 138.683,180.009 149.932,180.009 163.055,141.285  151.871,141.285 144.307,164.687 136.679,141.285 126.271,141.285 118.707,164.687 111.144,141.285 100.024,141.285  113.148,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="189.172,180.009 196.542,158.869 203.653,180.009 214.837,180.009 227.96,141.285  216.776,141.285 209.212,164.687 201.584,141.285 191.176,141.285 183.612,164.687 176.049,141.285 164.929,141.285  178.053,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="254.012,180.009 261.317,158.869 268.428,180.009 279.677,180.009 292.8,141.285  281.616,141.285 273.988,164.687 266.424,141.285 256.016,141.285 248.453,164.687 240.889,141.285 229.77,141.285  242.893,180.009  "
              style="fill: #194F82;"
            />
          </g>
        </svg>
      </span>
      <div
        class="vertical-timeline-element-content is-hidden"
        style="background: rgb(33, 150, 243); color: rgb(255, 255, 255);"
      >
        <div
          class="vertical-timeline-element-content-arrow"
          style="border-right: 7px solid rgb(33, 150, 243);"
        />
        <div
          data-testid="exp"
        >
          <h3
            class="vertical-timeline-element-title text-xl font-bold tracking-wide"
          >
            senior frontend developer
          </h3>
          <h4
            class="vertical-timeline-element-subtitle"
          >
            Avaaz Marketing corp
          </h4>
          <p>
            lorem ipsum dolor sit amet, consectetur adip proident et non proident consequat, sed do eiusmod tempor incididunt ut labore et dolore mag
          </p>
        </div>
        <span
          class="vertical-timeline-element-date"
        >
          Feb 2021 - Mar 2022
        </span>
      </div>
    </div>
    <div
      class="vertical-timeline-element--work vertical-timeline-element"
      id=""
    >
      <span
        class="vertical-timeline-element-icon is-hidden"
        style="background: rgb(138, 171, 255); color: rgb(255, 255, 255);"
      >
        <svg
          height="800px"
          id="Layer_1"
          viewBox="0 0 392.513 392.513"
          width="800px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <g>
            <path
              d="M329.261,256.421H63.37c0,0-39.434,83.329-39.628,83.588c-5.172,7.887,0.84,19.911,11.119,19.782 H357.77c9.438,0.453,16.743-10.925,11.119-19.782C368.695,339.75,329.261,256.421,329.261,256.421z"
              style="fill: #FFFFFF;"
            />
            <path
              d="M325.382,43.669c0-6.012-4.848-10.925-10.925-10.925H78.239c-6.012,0-10.925,4.848-10.925,10.925 v190.966h258.069V43.669z"
              style="fill: #FFFFFF;"
            />
          </g>
          <polygon
            points="48.954,338.005 77.204,278.207 315.426,278.207 343.742,338.005 "
            style="fill: #FFC10D;"
          />
          <path
            d="M89.099,54.595v13.059h39.176c6.012,0,10.925,4.848,10.925,10.925 c0,6.012-4.848,10.925-10.925,10.925H89.099v11.507h16.032c6.012,0,10.925,4.848,10.925,10.925c0,6.012-4.848,10.925-10.925,10.925 H89.164v90.053h214.432V54.595H89.099z"
            style="fill: #56ACE0;"
          />
          <g>
            <path
              d="M387.83,329.277l-40.663-85.98V43.669c0-18.036-14.675-32.711-32.711-32.711H78.239 c-18.036,0-32.711,14.675-32.711,32.711v199.564L4.8,329.213c-14.675,26.828,6.853,52.881,30.125,52.299H357.77 C378.586,382.74,402.893,357.269,387.83,329.277z M78.239,32.809h236.283c6.012,0,10.925,4.848,10.925,10.925V234.7H67.313V43.734 C67.313,37.657,72.162,32.809,78.239,32.809z M357.77,359.791H34.925c-10.279,0.129-16.356-11.895-11.119-19.782 c0.129-0.323,39.564-83.588,39.564-83.588h265.891c0,0,39.434,83.329,39.628,83.588C374.578,348.93,367.208,360.243,357.77,359.791 z"
              style="fill: #194F82;"
            />
            <path
              d="M225.826,310.918h-58.893c-6.012,0-10.925,4.849-10.925,10.925s4.848,10.925,10.925,10.925h58.893 c6.012,0,10.925-4.848,10.925-10.925C236.687,315.766,231.774,310.918,225.826,310.918z"
              style="fill: #194F82;"
            />
            <polygon
              points="124.267,180.009 131.637,158.869 138.683,180.009 149.932,180.009 163.055,141.285  151.871,141.285 144.307,164.687 136.679,141.285 126.271,141.285 118.707,164.687 111.144,141.285 100.024,141.285  113.148,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="189.172,180.009 196.542,158.869 203.653,180.009 214.837,180.009 227.96,141.285  216.776,141.285 209.212,164.687 201.584,141.285 191.176,141.285 183.612,164.687 176.049,141.285 164.929,141.285  178.053,180.009  "
              style="fill: #194F82;"
            />
            <polygon
              points="254.012,180.009 261.317,158.869 268.428,180.009 279.677,180.009 292.8,141.285  281.616,141.285 273.988,164.687 266.424,141.285 256.016,141.285 248.453,164.687 240.889,141.285 229.77,141.285  242.893,180.009  "
              style="fill: #194F82;"
            />
          </g>
        </svg>
      </span>
      <div
        class="vertical-timeline-element-content is-hidden"
        style="background: rgb(33, 150, 243); color: rgb(255, 255, 255);"
      >
        <div
          class="vertical-timeline-element-content-arrow"
          style="border-right: 7px solid rgb(33, 150, 243);"
        />
        <div
          data-testid="exp"
        >
          <h3
            class="vertical-timeline-element-title text-xl font-bold tracking-wide"
          >
            senior frontend engineer
          </h3>
          <h4
            class="vertical-timeline-element-subtitle"
          >
            Avaaztech corp
          </h4>
          <p>
            lorem ipsum dolor sit amet, consectetur adip proident et non proident consequat, sed do eiusmod tempor incididunt ut labore et dolore mag
          </p>
        </div>
        <span
          class="vertical-timeline-element-date"
        >
          Mar 2022 - Sep 2023
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
